### PR TITLE
Remove negative lookbehind from Rollup plugin that broke Node v9.11

### PR DESCRIPTION
**Pull request by bvaughn**
Only lightly tested 🤞😅

### CODE REVIEW

**Comment by react-linter** [comment]
React: size: -26.5%, gzip: -24.9%

ReactDOM: size: :small_red_triangle:+1.0%, gzip: :small_red_triangle:+0.5%


  <details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 28b9289022d1d9b646b99d0747eab904767ba5d6...931f1ab4b7b73d3fd264b2d2b12040c3e72f9601</p>


  
## react
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react.development.js | -31.2% | -30.1% | 83.89 KB | 57.7 KB | 22.76 KB | 15.91 KB | UMD_DEV
**react.production.min.js** | **-26.5%** | **-24.9%** | **9.59 KB** | **7.05 KB** | **3.97 KB** | **2.98 KB** | **UMD_PROD**
react.development.js | -10.1% | -8.9% | 53.77 KB | 48.34 KB | 14.88 KB | 13.56 KB | NODE_DEV
**react.production.min.js** | **-6.7%** | **-5.6%** | **5.99 KB** | **5.59 KB** | **2.57 KB** | **2.42 KB** | **NODE_PROD**
React-dev.js | -5.5% | -6.7% | 51.38 KB | 48.56 KB | 14.22 KB | 13.26 KB | FB_WWW_DEV
**React-prod.js** | **-3.8%** | **-3.7%** | **13.91 KB** | **13.39 KB** | **3.89 KB** | **3.75 KB** | **FB_WWW_PROD**

## react-dom
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-dom.development.js | -3.1% | -3.6% | 646.54 KB | 626.47 KB | 151.28 KB | 145.79 KB | UMD_DEV
**react-dom.production.min.js** | **:small_red_triangle:+1.0%** | **:small_red_triangle:+0.5%** | **93.32 KB** | **94.25 KB** | **30.37 KB** | **30.53 KB** | **UMD_PROD**
react-dom.development.js | -4.8% | -5.4% | 641.85 KB | 610.84 KB | 149.91 KB | 141.79 KB | NODE_DEV
**react-dom.production.min.js** | **-0.6%** | **-2.0%** | **93.28 KB** | **92.74 KB** | **30.13 KB** | **29.52 KB** | **NODE_PROD**
react-dom-test-utils.development.js | +0.4% | +2.0% | 45.07 KB | 45.27 KB | 12.23 KB | 12.47 KB | UMD_DEV
**react-dom-test-utils.production.min.js** | **-0.9%** | **-1.1%** | **10.5 KB** | **10.4 KB** | **3.87 KB** | **3.83 KB** | **UMD_PROD**
react-dom-test-utils.development.js | -10.4% | -9.2% | 44.79 KB | 40.13 KB | 12.16 KB | 11.04 KB | NODE_DEV
**react-dom-test-utils.production.min.js** | **-6.0%** | **-5.0%** | **10.3 KB** | **9.68 KB** | **3.8 KB** | **3.61 KB** | **NODE_PROD**
react-dom-unstable-native-dependencies.development.js | +2.0% | +2.6% | 60.26 KB | 61.45 KB | 15.76 KB | 16.17 KB | UMD_DEV
**react-dom-unstable-native-dependencies.production.min.js** | **:small_red_triangle:+1.8%** | **:small_red_triangle:+2.2%** | **11.15 KB** | **11.35 KB** | **3.83 KB** | **3.91 KB** | **UMD_PROD**
react-dom-unstable-native-dependencies.development.js | -4.7% | -4.6% | 59.93 KB | 57.11 KB | 15.63 KB | 14.9 KB | NODE_DEV
**react-dom-unstable-native-dependencies.production.min.js** | **-2.0%** | **-2.3%** | **10.9 KB** | **10.68 KB** | **3.74 KB** | **3.66 KB** | **NODE_PROD**
react-dom-server.browser.development.js | -1.4% | -3.4% | 104.02 KB | 102.54 KB | 27.83 KB | 26.88 KB | UMD_DEV
**react-dom-server.browser.production.min.js** | **-1.5%** | **-1.4%** | **15.33 KB** | **15.1 KB** | **5.86 KB** | **5.78 KB** | **UMD_PROD**
react-dom-server.browser.development.js | -8.3% | -8.3% | 100.15 KB | 91.85 KB | 26.85 KB | 24.63 KB | NODE_DEV
**react-dom-server.browser.production.min.js** | **-5.0%** | **-4.8%** | **15.23 KB** | **14.46 KB** | **5.81 KB** | **5.53 KB** | **NODE_PROD**
react-dom-server.node.development.js | -8.1% | -8.1% | 102.07 KB | 93.77 KB | 27.38 KB | 25.16 KB | NODE_DEV
**react-dom-server.node.production.min.js** | **-4.8%** | **-4.7%** | **16.04 KB** | **15.27 KB** | **6.11 KB** | **5.82 KB** | **NODE_PROD**
ReactDOM-dev.js | -6.6% | -7.1% | 663.63 KB | 620.02 KB | 151.92 KB | 141.13 KB | FB_WWW_DEV
**ReactDOM-prod.js** | **-6.4%** | **-5.1%** | **287.71 KB** | **269.43 KB** | **53.3 KB** | **50.56 KB** | **FB_WWW_PROD**
ReactTestUtils-dev.js | -0.9% | +0.2% | 41.88 KB | 41.52 KB | 11.21 KB | 11.24 KB | FB_WWW_DEV
ReactDOMUnstableNativeDependencies-dev.js | -0.5% | +0.8% | 57.63 KB | 57.32 KB | 14.58 KB | 14.69 KB | FB_WWW_DEV
**ReactDOMUnstableNativeDependencies-prod.js** | **-1.3%** | **:small_red_triangle:+0.3%** | **26.59 KB** | **26.25 KB** | **5.32 KB** | **5.34 KB** | **FB_WWW_PROD**
ReactDOMServer-dev.js | -5.6% | -7.6% | 101.25 KB | 95.57 KB | 26.51 KB | 24.51 KB | FB_WWW_DEV
**ReactDOMServer-prod.js** | **-4.6%** | **-3.3%** | **33.06 KB** | **31.54 KB** | **8 KB** | **7.74 KB** | **FB_WWW_PROD**
react-dom.profiling.min.js | -2.6% | -2.9% | 96.14 KB | 93.65 KB | 30.75 KB | 29.86 KB | NODE_PROFILING

## react-art
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-art.development.js | -7.0% | -7.4% | 438.74 KB | 408.05 KB | 98.28 KB | 91.03 KB | UMD_DEV
**react-art.production.min.js** | **-4.1%** | **-3.7%** | **84.58 KB** | **81.12 KB** | **25.98 KB** | **25.01 KB** | **UMD_PROD**
react-art.development.js | -9.9% | -11.2% | 370.46 KB | 333.9 KB | 81.19 KB | 72.12 KB | NODE_DEV
**react-art.production.min.js** | **-8.2%** | **-7.2%** | **49.54 KB** | **45.48 KB** | **15.27 KB** | **14.16 KB** | **NODE_PROD**
ReactART-dev.js | -13.1% | -15.0% | 375.41 KB | 326.2 KB | 79.97 KB | 68.01 KB | FB_WWW_DEV
**ReactART-prod.js** | **-11.1%** | **-10.6%** | **159.88 KB** | **142.16 KB** | **27.12 KB** | **24.25 KB** | **FB_WWW_PROD**

## react-test-renderer
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-test-renderer.development.js | -10.8% | -11.9% | 382.52 KB | 341.05 KB | 83.77 KB | 73.8 KB | UMD_DEV
**react-test-renderer.production.min.js** | **-9.5%** | **-8.3%** | **50.75 KB** | **45.92 KB** | **15.51 KB** | **14.22 KB** | **UMD_PROD**
react-test-renderer.development.js | -12.2% | -14.0% | 378.01 KB | 331.88 KB | 82.64 KB | 71.07 KB | NODE_DEV
**react-test-renderer.production.min.js** | **-10.8%** | **-10.3%** | **50.46 KB** | **45.03 KB** | **15.35 KB** | **13.76 KB** | **NODE_PROD**
react-test-renderer-shallow.development.js | +0.3% | -1.6% | 24.29 KB | 24.36 KB | 6.61 KB | 6.5 KB | UMD_DEV
**react-test-renderer-shallow.production.min.js** | **-0.3%** | **-0.2%** | **7.17 KB** | **7.15 KB** | **2.35 KB** | **2.34 KB** | **UMD_PROD**
react-test-renderer-shallow.development.js | -26.1% | -33.5% | 19.48 KB | 14.41 KB | 5.42 KB | 3.61 KB | NODE_DEV
**react-test-renderer-shallow.production.min.js** | **-8.6%** | **-10.2%** | **7.87 KB** | **7.19 KB** | **2.61 KB** | **2.35 KB** | **NODE_PROD**
ReactTestRenderer-dev.js | -11.8% | -13.5% | 383.12 KB | 337.73 KB | 81.67 KB | 70.65 KB | FB_WWW_DEV
ReactShallowRenderer-dev.js | -14.9% | -20.5% | 17.79 KB | 15.14 KB | 4.69 KB | 3.73 KB | FB_WWW_DEV

## react-noop-renderer
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-noop-renderer.development.js | -23.2% | -12.0% | 23.35 KB | 17.93 KB | 5.33 KB | 4.69 KB | NODE_DEV
**react-noop-renderer.production.min.js** | **-23.8%** | **-15.4%** | **8.75 KB** | **6.67 KB** | **3.04 KB** | **2.57 KB** | **NODE_PROD**
react-noop-renderer-persistent.development.js | -23.1% | -11.9% | 23.46 KB | 18.05 KB | 5.34 KB | 4.71 KB | NODE_DEV
**react-noop-renderer-persistent.production.min.js** | **-23.7%** | **-15.3%** | **8.77 KB** | **6.69 KB** | **3.04 KB** | **2.58 KB** | **NODE_PROD**

## react-reconciler
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-reconciler.development.js | -11.5% | -13.5% | 366.29 KB | 324.11 KB | 79.3 KB | 68.62 KB | NODE_DEV
**react-reconciler.production.min.js** | **-8.1%** | **-8.6%** | **49.1 KB** | **45.15 KB** | **14.76 KB** | **13.49 KB** | **NODE_PROD**
react-reconciler-persistent.development.js | -11.5% | -13.6% | 364.85 KB | 322.73 KB | 78.73 KB | 68.04 KB | NODE_DEV
**react-reconciler-persistent.production.min.js** | **-8.1%** | **-8.6%** | **49.11 KB** | **45.16 KB** | **14.76 KB** | **13.49 KB** | **NODE_PROD**
react-reconciler-reflection.development.js | -21.6% | -22.9% | 14.76 KB | 11.57 KB | 4.62 KB | 3.56 KB | NODE_DEV
**react-reconciler-reflection.production.min.js** | **-11.6%** | **-11.5%** | **2.55 KB** | **2.26 KB** | **1.13 KB** | **1 KB** | **NODE_PROD**

## simple-cache-provider
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
simple-cache-provider.development.js | -15.9% | -17.8% | 8.89 KB | 7.47 KB | 2.86 KB | 2.36 KB | NODE_DEV
**simple-cache-provider.production.min.js** | **-0.5%** | **-0.7%** | **1.64 KB** | **1.63 KB** | **833 B** | **827 B** | **NODE_PROD**
SimpleCacheProvider-dev.js | -0.6% | -0.2% | 7.92 KB | 7.87 KB | 2.4 KB | 2.39 KB | FB_WWW_DEV
**SimpleCacheProvider-prod.js** | **-0.5%** | **-0.9%** | **3.65 KB** | **3.63 KB** | **1.11 KB** | **1.1 KB** | **FB_WWW_PROD**

## create-subscription
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
create-subscription.development.js | -31.2% | -30.5% | 8 KB | 5.5 KB | 2.77 KB | 1.93 KB | NODE_DEV
**create-subscription.production.min.js** | **-10.3%** | **-8.9%** | **2.82 KB** | **2.53 KB** | **1.32 KB** | **1.2 KB** | **NODE_PROD**

## react-native-renderer
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
ReactNativeRenderer-dev.js | -8.1% | -9.2% | 497.7 KB | 457.61 KB | 110.15 KB | 100.01 KB | RN_FB_DEV
**ReactNativeRenderer-prod.js** | **-4.1%** | **-4.1%** | **214.75 KB** | **205.92 KB** | **37.45 KB** | **35.92 KB** | **RN_FB_PROD**
ReactNativeRenderer-dev.js | -8.1% | -9.2% | 497.44 KB | 457.32 KB | 110.08 KB | 99.95 KB | RN_OSS_DEV
**ReactNativeRenderer-prod.js** | **-5.1%** | **-5.2%** | **204.57 KB** | **194.12 KB** | **35.8 KB** | **33.94 KB** | **RN_OSS_PROD**
ReactFabric-dev.js | -8.1% | -9.2% | 487.89 KB | 448.58 KB | 107.73 KB | 97.78 KB | RN_FB_DEV
**ReactFabric-prod.js** | **-5.4%** | **-5.0%** | **197.07 KB** | **186.37 KB** | **34.31 KB** | **32.59 KB** | **RN_FB_PROD**
ReactFabric-dev.js | -8.1% | -9.2% | 487.92 KB | 448.61 KB | 107.75 KB | 97.79 KB | RN_OSS_DEV
**ReactFabric-prod.js** | **-5.4%** | **-5.0%** | **197.1 KB** | **186.4 KB** | **34.32 KB** | **32.61 KB** | **RN_OSS_PROD**
ReactNativeRenderer-profiling.js | -7.1% | -7.2% | 211.73 KB | 196.66 KB | 37.18 KB | 34.48 KB | RN_OSS_PROFILING
ReactFabric-profiling.js | -7.1% | -7.2% | 203.06 KB | 188.62 KB | 35.62 KB | 33.07 KB | RN_OSS_PROFILING

## react-scheduler
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-scheduler.development.js | +24.4% | +23.6% | 15.4 KB | 19.17 KB | 4.65 KB | 5.74 KB | UMD_DEV
**react-scheduler.production.min.js** | **:small_red_triangle:+15.8%** | **:small_red_triangle:+21.0%** | **2.73 KB** | **3.16 KB** | **1.26 KB** | **1.53 KB** | **UMD_PROD**
react-scheduler.development.js | -8.1% | -7.9% | 15.35 KB | 14.11 KB | 4.62 KB | 4.25 KB | NODE_DEV
**react-scheduler.production.min.js** | **-1.0%** | **:small_red_triangle:+11.0%** | **2.79 KB** | **2.76 KB** | **1.22 KB** | **1.35 KB** | **NODE_PROD**
ReactScheduler-dev.js | -9.4% | -8.5% | 15.57 KB | 14.11 KB | 4.66 KB | 4.26 KB | FB_WWW_DEV
**ReactScheduler-prod.js** | **-1.3%** | **:small_red_triangle:+8.1%** | **7.72 KB** | **7.62 KB** | **1.9 KB** | **2.06 KB** | **FB_WWW_PROD**

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by gaearon** [comment]
```
Error: Error transforming bundle with 'rollup-plugin-prettier' plugin: Unexpected token, expected ";" (4:82)
  2 |  Modernizr 3.0.0pre (Custom Build) | MIT
  3 | */
> 4 | 'use strict';var React=require("React"),reactScheduler=require("react-scheduler")var invariant=require("invariant");
    |                                                                                  ^
```

File: scripts/rollup/plugins/strip-unused-imports.js [hunk 1]
```diff
@@ -11,8 +11,14 @@ module.exports = function sizes(pureExternalModules) {
     name: 'scripts/rollup/plugins/strip-unused-imports',
     transformBundle(code) {
       pureExternalModules.forEach(module => {
+        // Ideally this would use a negative lookbehind: (?<!= *)
+        // But this isn't supported by the version of Node that Prepack runs fasted on,
```
Comment (gaearon) [inline scripts/rollup/plugins/strip-unused-imports.js]: "faster"? :-)

I think this clarification taken out of context is more confusing than it's worth.

File: scripts/rollup/plugins/strip-unused-imports.js [hunk 2]
```diff
@@ -11,8 +11,14 @@ module.exports = function sizes(pureExternalModules) {
...
+        // So instead we try to handle the most common cases:
+        // 1. foo,bar=require("bar"),baz
+        // 2. foo;bar = require('bar');baz
+        // 3.   require('bar');
         const regExp = new RegExp(
-          `(?<!= *)require\\(["']${module}["']\\)[,;]`,
+          `([,;]| {2})require\\(["']${module}["']\\)[,;]`,
           'g'
         );
         code = code.replace(regExp, '');
```
[the remark this answers is not in the document]
Comment (bvaughn) [reply scripts/rollup/plugins/strip-unused-imports.js]: I'm not sure I understand what you're asking? The plugin name is "strip-unused-imports"

**Comment by bvaughn** [comment]
Whoops. That's right, I need to replace with the $1 group since I'm not using the lookbehind.

File: scripts/rollup/plugins/strip-unused-imports.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (gaearon) [reply scripts/rollup/plugins/strip-unused-imports.js]:
```js
module.exports = function sizes(
```

**Comment by bvaughn** [comment]
Ok. Back from shopping and actually had the time to run the full build script and test a couple of fixtures. Check again?

**Comment by gaearon** [comment]
What happened to the PR

**Comment by bvaughn** [comment]
😕 uh...

**Comment by bvaughn** [comment]
Force push gone wrong I guess

**Comment by gaearon** [comment]
Let's wait for bundle size change comment to disappear

**Comment by bvaughn** [comment]
I felt better about the previous form of the regex 😄but...

Worst case, we _could_ just replace this with a hard-coded string replace _specifically_ targeted at the tracking import. If this _did_ break the bundle, our packaging fixture would catch it I think?

**Comment by nhunzaker** [comment]
Comment disappeared. But yeah.

**Comment by gaearon** [comment]
>But if this did break the bundle, our packaging fixture would catch it I think?

Hopefully.

**Pull request merged**